tests/s3/copy_object/test_copy_object_quota.py

```diff
@@ -0,0 +1,733 @@
+        self.log.info("Step 2: Perform PUT API to set user level quota fields i.e."
+                      "enabled(bool)=true, max_size(integer value)= -1 ,"
+                      " max_objects(integer value) = N ")
+        self.set_get_user_quota(config_dict, self.user_id)
+        self.log.info("Step 3:Perform GET API to get user level quota fields and verify the"
+                      " user level quota fields as per above PUT request.")
+        self.verify_user_quota(self.akey, self.skey, self.user_id)
+        self.log.info("Step 4: Create 'N+1' buckets named 'src1' , 'dest1' .. 'destN' under the"
+                      "user created in step 1")
+        for cnt in range(1, test_cfg["max_objects"]+2):
+            bkt_name = "dest" + str(cnt)
+            bucket_created = s3_misc.create_bucket(bkt_name, self.akey, self.skey)
+            assert_utils.assert_true(bucket_created, "Failed to create bucket")
+            self.buckets_created.append([bkt_name, self.akey, self.skey])
+        self.log.info("Step 5: Upload a simple object obj1 of some random size S in src1")
+        random_size = self.csm_obj.random_gen.randrange(1, test_cfg["max_size"])
+        resp = self.multipart_upload(self.src_bkt, "obj1", self.akey, self.skey, random_size)
+        assert_utils.assert_true(resp, "Multipart upload Failed")
+        self.log.info("Step 6: Perform Copy object operation within same bucket 'N-1' times using"
+                      "same source bucket and object and just changing the key name of the "
+                      "destination object .")
+        for cnt in range(1, (test_cfg["max_objects"])):
+            dest_bkt = "dest" + str(cnt)
+            resp = s3_misc.copy_object(self.akey, self.skey, self.src_bkt, "obj1",
+                                       dest_bkt, "obj1")
+            assert_utils.assert_true(resp, f"Copy object Failed for {dest_bkt}/obj1")
+            self.src_bkt = dest_bkt
+        self.log.info("Step 7: Perform Copy object operation again (N)th time from 'destN-1'"
+                      "to 'destN' bucket and keeping the key name of the destination object"
+                      "same during every copy operation.")
+        try:
+            dest_bkt = "dest" + str(test_cfg["max_objects"])
+            s3_misc.copy_object(self.akey, self.skey, self.src_bkt, "obj1", dest_bkt, "obj1")
+        except CTException as error:
+            self.log.info("Expected exception received %s", error)
+            assert_utils.assert_in(errmsg.S3_COPY_OBJECT_QUOTA_ERR, error.message, error)
+        self.log.info("Step 8:Perform Get API to get user and bucket stats and validate the object"
+                      "count and space utilization for user/bucket .")
+        self.log.debug("Perform & Verify GET API to get capacity usage stats")
+        self.verify_user_quota(self.akey, self.skey, self.user_id)
+        self.log.info("Step 9: Create one more user (user2)")
```

Comment:
can we have for loop for user 1 and 2, to avoid code duplicacy

Reply:
we cannot, as explained above